File: freecodecamp/daily-code-challenge/2026-02-27.js
function shiftMatrix(matrix, shift) {
    if (!matrix.length || !matrix[0].length) {
        return matrix;
    }

    const rows = matrix.length;
    const cols = matrix[0].length;
    const total = rows * cols;

    // Normalize shift
    shift = ((shift % total) + total) % total;

    // Flatten matrix
    const flat = matrix.flat();

    // Shift using slicing
    const shifted = flat.slice(-shift).concat(flat.slice(0, total - shift));

    // Rebuild matrix
    const result = [];
    for (let r = 0; r < rows; r++) {
        result.push(shifted.slice(r * cols, (r + 1) * cols));
    }

    return result;
}

File: freecodecamp/daily-code-challenge/2026-02-27.py
def shift_matrix(matrix, shift):
    if not matrix or not matrix[0]:
        return matrix

    rows = len(matrix)
    cols = len(matrix[0])
    total = rows * cols

    # Normalize shift
    shift %= total

    # Flatten matrix
    flat = [matrix[r][c] for r in range(rows) for c in range(cols)]

    # Shift using slicing
    shifted = flat[-shift:] + flat[:-shift]

    # Rebuild matrix
    result = []
    for r in range(rows):
        row = shifted[r * cols:(r + 1) * cols]
        result.append(row)

    return result
